### ISSUE TITLE
extractfeat: option to add attributes from 9th column to defline of resulting FASTA

### ISSUE THREAD

**Issue by yeban** [genometools/genometools#571]
Much like `-seqid` option, it would be great if other attributes like 'Name', 'Dbxref', etc. can be added to the defline of the extracted sequence.

**Comment by gordon** [comment]
What about adding an `-attr` option which is implemented as a `gt_option_new_string_array()` option and which allows to specify all the attributes which should be added to the extracted sequence description?
The attribute values do not necessarily have to be the same, so that should be taken into account (similar to option `-target`).

Option `-attr` would be flexible enough for all attribute needs and we would not have to add another option for each attribute we want to add to the description.

@yeban: Do you want to work on this? Please send a pull request. If you run into problems, feel free to ask me here.

**Comment by satta** [comment]
I completely agree (I have been doing something similar separately up to now).
So in case of non-joined features (where one feature corresponds to one sequence) the attribute values are taken from the extracted feature itself, and in the joined case (e.g. CDS joined via their parent transcripts) the attributes of the parent are used?

**Comment by yeban** [comment]
@gordon I will sure give it a shot as soon as I can find some time. I guess looking at how `-retainds` words will be a good start.

At the moment I'm tweaking feature ids in the GFF to contain all relevant info and using `-retainids` switch later :P.

@satta Picking attribute from parent would be great when using `-join` (which is what I always use). Shouldn't `-retainds`do the same?

**Comment by satta** [comment]
@yeban The `-retainids` option in `gt extractfeat` should not change the behaviour of `-join`, it would just make sure that the 'original' IDs from the input GFF3 are used to label the sequences. So I would expect it to label joined sequences with the ID of the parent.

**Comment by gordon** [comment]
@satta: The approach taken for option `-target` in the case of joined features is to add all target IDs to the description. I'm not sure which approach would be better for attributes: Join all different attributes from the children or take the attribute of the parent.

**Comment by satta** [comment]
@gordon I guess one could implement both approaches and take the parent attribute by default?

**Comment by gordon** [comment]
Sounds good.

**Comment by sjackman** [comment]
:+1:

**Comment by sjackman** [comment]
@yeban Bump?

**Comment by yeban** [comment]
@sjackman Sorry, I pass. I maintain that this issue or at least #605 is very relevant to me, but I'm not able to make time.

**Comment by sjackman** [comment]
No worries. I don't have time right now either. Perhaps in about a month.

**Comment by photocyte** [comment]
Hi there, was this ever addressed?  I'd also be interested in this feature & the latest version of gt doesn't appear to have it.

**Comment by satta** [comment]
No, in fact is hasn't been addressed yet. Looks like interest has faded a bit. Any volunteers? :)
I could take a look at how long it would take to implement this but my time is unfortunately _very_ limited atm.

**Comment by photocyte** [comment]
I took a quick look at the (presumed) [source](https://github.com/genometools/genometools/blob/ba61a9536c5e1b1f2f2141f3a0816fe8d5c3fb74/src/tools/gt_extractfeat.c).  I'm not totally ignorant of C, but it definitely isn't my strongest suit, and looking at the source, it is definitely not obvious to me what is going on. Mainly, I want to make it so the FASTA record description produced by extractfeat from a joined CDS has the GFF Name attribute from the mRNA (or the gene, if the mRNA Name attribute isn't available).  I could try to stare at that C file I linked above to try and figure out how to add it... If it takes more modifications in other source files to get the arguments properly printed / parsed, let me know as that would greatly help any attempt to modify.

**Comment by satta** [comment]
Correct code file, at least one of them :) I will need to take a closer look myself to give more direction, as I haven't looked at that code for quite some time as well now. I'll try to get back to you in the next couple of days, OK?
